pytorch_lightning/trainer/trainer.py

```diff
@@ -1494,6 +1500,42 @@ def call_hook(
 
         return output
 
+    def _profile_and_call(self, obj: object, hook_name: str, *args: Any, **kwargs: Any) -> Optional[Any]:
+        fn = getattr(obj, hook_name)
+        with self.profiler.profile(f"{obj.__class__.__name__}.{hook_name}"):
+            return fn(*args, **kwargs)
+
+    def _call_hook(
+        self,
+        obj: object,
+        hook_name: str,
+        *args: Any,
+        pl_module: Optional["pl.LightningModule"] = None,
+        **kwargs: Any,
+    ) -> Optional[Any]:
+        pl_module = pl_module or self.lightning_module
+
+        if pl_module is not None:
+            prev_fx_name = pl_module._current_fx_name
+            pl_module._current_fx_name = hook_name
+
+        output = None
+        if isinstance(obj, Trainer):
+            for obj in self.callbacks:
+                if hook_name in ("on_init_start", "on_init_end"):
+                    # these `Callback` hooks are the only ones that do not take a lightning module
+                    output = self._profile_and_call(obj, hook_name, self, *args, **kwargs)
+                else:
+                    output = self._profile_and_call(obj, hook_name, self, pl_module, *args, **kwargs)
```

Comment:
callbacks have a `state_key` which we should use to distinguish entries in the profiler output: https://github.com/PyTorchLightning/pytorch-lightning/blob/af0bb96f0ff645102680a7adc99dc131cfeb9c0b/pytorch_lightning/callbacks/base.py#L36-L44

Reply:
Good call, added